sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -36,14 +36,19 @@
             .identityClientOptions(identityClientOptions)
             .build();
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
-        if (configuration.contains(Configuration.PROPERTY_MSI_ENDPOINT)
-                || (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)
-                        && configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER))) {
-            appServiceMSICredential = new AppServiceMsiCredential(clientId, identityClient);
-            virtualMachineMSICredential = null;
+        if (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)) {
+            if (configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER)) {
+                managedIdentityServiceCredential = new AppServiceMsiCredential(clientId, identityClient);
+            } else if (configuration.contains(PROPERTY_IMDS_ENDPOINT)) {
```

Comment:
Environment variable `IMDS_ENDPOINT` is never used in `ArcIdentityCredential`, is that normal?

Reply:
If so - what's the value for?

Reply:
Currently, its just used to determine the Arc Env.
After discussing with @catalinaperalta , she mentioned we won't try auth with their IMDS endpoint.

Reply:
This is correct we only use `IMDS_ENDPOINT` as a signal to indicate that were on an Arc host.